packages/contracts/deploy/new/40_finalize-managing-dao/99_verify_step.ts

```diff
@@ -46,20 +45,6 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
     permission: 'ROOT_PERMISSION',
   });
 
-  await checkPermission(managingDaoContract, {
```

Comment:
I just looked inside `checkPermission` which calls `isPermissionSetCorrectly`. 
I think Mathias didn't cover the case `GrantWithCondition` case in `isPermissionSetCorrectly` properly.
Since GeoBrowser might use this a lot, we should check and potentially fix it.
Maybe in a separate PR, but please give it a look too.

Reply:
https://github.com/aragon/osx/blob/08a8ff89bfc1ee601a26d5a88ebe5fe093336a4d/packages/contracts/deploy/helpers.ts#L378-L385

Reply:
I don't think this is needed. `checkPermission` is only called for hardcoded permissions that we use in deploy script. This function isn't used for permissions set inside plugin setups at all. Once geo-browser deploys it, they don't touch our osx, so we're good for now. Good finding, but not important for now, until we need to write another `checkPermission` manually with `grantWithCondition`.

Reply:
Let's make a task please @Rekard0

Reply:
Thanks, https://aragonassociation.atlassian.net/browse/OS-1232